src/include/miopen/tensor_layout.hpp

```diff
@@ -62,18 +62,21 @@ void tensor_layout_to_strides(const std::vector<T>& len,
                        }
                        return std::accumulate(layout.begin() + pos + 1,
                                               layout.end(),
-                                              1,
+                                              static_cast<T>(1),
                                               [&dim_to_len](T accumulator, char l) {
                                                   return accumulator * dim_to_len[l];
                                               });
                    });
 }
 
+/// \brief Version for vectorized layouts.
+///
+/// \todo Generalize with non-vectorized version, 90% of code is the same.
 template <typename T>
 void tensor_layout_to_strides(const std::vector<T>& len,
                               const std::string& len_layout,
                               const std::string& layout,
-                              const int vector,
+                              const T vector_size,
```

Comment:
This is because how this function is called. When `tensor_layout_to_strides` is called with `T = size_t`, this argument passed to `vector_size` is also `size_t`. That leads to warning because `size_t` is implicitly casted to `int`. Changed to `T` to fix the warnings.

Reply:
Do we have this warning enabled? We just have a lot of places where this kind of implicit conversion happens. Also here we can always use `std::size_t` because that's what the `TensorDescriptor::GetVectorLength()` gives us.

Reply:
> Do we have this warning enabled?

IIRC I've seen it during normal builds (not tidy) with ROCm 6.0.

> Also here we can always use `std::size_t` because that's what the `TensorDescriptor::GetVectorLength()` gives us.

Indeed.

Reply:
Fixed in 6dd37fb8f